irc/client.go

```diff
@@ -84,6 +84,11 @@ SelectLoop:
 
 		default:
 			if !s.Scan() {
+				if err := s.Err(); err == nil {
```

Comment:
I think we should unconditionally shutdown this client connection if we run into an error. I tried this commit and it doesn't seem to pick up on closed connection unless we always `close` on `!s.Scan()`